zebra-chain/src/serialization/constraint.rs

```diff
@@ -178,3 +178,33 @@ impl<T> AtLeastOne<T> {
 
 // TODO: consider implementing `push`, `append`, and `Extend`,
 // because adding elements can't break the constraint.
+
+/// Create an initialized [`AtLeastOne`] instance.
+///
+/// This macro is similar to the [`vec!`][`std::vec!`] macro, but doesn't support creating an empty
+/// `AtLeastOne` instance.
+///
+/// Note: This is currently only used for testing.
```

Comment:
```suggestion
/// # Security
///
/// This macro must only be used in tests, because it skips the `TrustedPreallocate` memory
/// denial of service checks.
```